xtractmime/__init__.py

```diff
@@ -0,0 +1,76 @@
+from typing import List, Union, Tuple
+
+_APACHE_TYPES = [
+    b"text/plain",
+    b"text/plain; charset=ISO-8859-1",
+    b"text/plain; charset=iso-8859-1",
+    b"text/plain; charset=UTF-8",
+]
+
+
+_WHITESPACE_BYTES = {b"\t", b"\r", b"\x0c", b"\n", b" "}
+
+
+# handling resource metadata
+def _should_check_for_apache_bug(supplied_type, http_origin):
+    return http_origin and supplied_type in _APACHE_TYPES
+
+
+# reading resource header
+def _read_resource_header(body):
+    buffer = memoryview(body)
+    if len(body) < 1445:
+        return buffer
+    else:
+        return buffer[:1445]
+    
+
+
+# Matching a MIME type pattern
+def _is_match_mime_pattern(
+    input_bytes, byte_pattern, pattern_mask, lead_whitespace=None
+):
+    input_size = len(input_bytes)
+    pattern_size = len(byte_pattern)
+    mask_size = len(pattern_mask)
+
+    if pattern_size != mask_size:
+        raise ValueError("pattern's length should match mask's length")
+
+    if input_size < pattern_size:
+        return False
+
+    input_index, pattern_index = 0, 0
+
+    if lead_whitespace:
+        while (
+            input_index < input_size
+            and input_bytes[input_index:input_index+1] in _WHITESPACE_BYTES
+        ):
+            input_index += 1
+
+    while pattern_index < pattern_size:
+        masked_byte = bytes([input_bytes[input_index] & pattern_mask[pattern_index]])
```

Comment:
Would this work?

```suggestion
        masked_byte = input_bytes[input_index:input_index+1] & pattern_mask[pattern_index:pattern_index+1]
```

Reply:
No, it's giving this error `TypeError: unsupported operand type(s) for &: 'bytes' and 'bytes'`

Reply:
[Looks like it can’t be helped](https://stackoverflow.com/a/22593976). I’m learning a lot :slightly_smiling_face: